SteamKit2/SteamKit2/Steam/CDN/Client.cs

```diff
@@ -227,10 +232,31 @@ public async Task<int> DownloadDepotChunkAsync( uint depotId, DepotManifest.Chun
                 }
             }
 
+            await _lancacheDetectionLock.WaitAsync();
```

Comment:
Putting this check inside of depot download wont perform for manifest downloads.

But my initial idea was to add this as a separate public method so consumers can chose to check/enable for lancache support, instead of it happening automagically.

Reply:
This is a good thing to be honest. Since manifests are downloaded with request codes appended to the URL, if they were to be passed through lancache, it could in theory make the same manifest be downloaded and saved multiple times, as lancache uses the complete url string to tell if it's already saved or a new file.
In short, new request code = new url = new file.

Reply:
Does Steam forward manifest requests to lancache?

Reply:
Yes it does.

Reply:
It appears so. I checked my access logs in lancache, and it seems that even the Steamdeck, if it detects lancache, will send the request through there. I have snippet saved to an ods spreadsheet if you want to review it. Can send it to you via Discord.